Plugins/NativeInput/Source/Shared/NativeInput.cpp

```diff
@@ -40,55 +42,112 @@ namespace Babylon::Plugins
         return *env.Global().Get(JS_NATIVE_INPUT_NAME).As<Napi::External<NativeInput>>().Data();
     }
 
-    void NativeInput::PointerDown(uint32_t pointerId, uint32_t buttonIndex, uint32_t x, uint32_t y)
+    void NativeInput::MouseDown(uint32_t buttonIndex, uint32_t x, uint32_t y)
     {
-        m_impl->PointerDown(pointerId, buttonIndex, x, y);
+        m_impl->MouseDown(buttonIndex, x, y);
     }
 
-    void NativeInput::PointerUp(uint32_t pointerId, uint32_t buttonIndex, uint32_t x, uint32_t y)
+    void NativeInput::MouseUp(uint32_t buttonIndex, uint32_t x, uint32_t y)
     {
-        m_impl->PointerUp(pointerId, buttonIndex, x, y);
+        m_impl->MouseUp(buttonIndex, x, y);
     }
 
-    void NativeInput::PointerMove(uint32_t pointerId, uint32_t x, uint32_t y)
+    void NativeInput::MouseMove(uint32_t x, uint32_t y)
     {
-        m_impl->PointerMove(pointerId, x, y);
+        m_impl->MouseMove(x, y);
+    }
+
+    void NativeInput::TouchDown(uint32_t pointerId, uint32_t buttonIndex, uint32_t x, uint32_t y)
+    {
+        m_impl->TouchDown(pointerId, buttonIndex, x, y);
+    }
+
+    void NativeInput::TouchUp(uint32_t pointerId, uint32_t buttonIndex, uint32_t x, uint32_t y)
+    {
+        m_impl->TouchUp(pointerId, buttonIndex, x, y);
+    }
+
+    void NativeInput::TouchMove(uint32_t pointerId, uint32_t x, uint32_t y)
+    {
+        m_impl->TouchMove(pointerId, x, y);
     }
 
     NativeInput::Impl::Impl(Napi::Env env)
         : m_runtimeScheduler{JsRuntime::GetFromJavaScript(env)}
     {
         NativeInput::Impl::DeviceInputSystem::Initialize(env);
+
+        if (HasMouse())
+        {
+            const uint32_t inputIndex{ GetPointerButtonInputIndex(0) };
+            GetOrCreateInputMap(DeviceType::Mouse, MOUSE_POINTER_ID, { inputIndex, POINTER_X_INPUT_INDEX, POINTER_Y_INPUT_INDEX });
```

Comment:
Is this just so that we can query these inputs before there are any mouse events?

Reply:
yes, that was the thought behind it